lib/improver/metadata/check_datatypes.py

```diff
@@ -0,0 +1,222 @@
+
+
+def check_cube_not_float64(cube, fix=False):
+    """Check a cube does not contain any float64 data, excepting time
+    coordinates. The cube can be modified in place, if the fix keyword is
+    specified to be True.
+
+    Args:
+        cube (iris.cube.Cube):
+            The input cube that will be checked for float64 inclusion.
+        fix (bool):
+            If fix is True, then the cube is amended to not include float64
+            data, otherwise, an error will be raised if float64 data is found.
+
+    Raises:
+        TypeError : Raised if float64 values are found in the cube.
+
+    """
+    if cube.dtype == np.float64:
+        if fix:
+            cube.data = cube.data.astype(np.float32)
+        else:
+            raise TypeError("64 bit cube not allowed: {!r}".format(cube))
+    for coord in cube.coords():
+        if coord.name() in ["time", "forecast_reference_time"]:
+            continue
+        if coord.points.dtype == np.float64:
+            if fix:
+                coord.points = coord.points.astype(np.float32)
+            else:
+                raise TypeError(
+                    "64 bit coord points not allowed: {} in {!r}".format(
+                        coord, cube))
+        if (hasattr(coord, "bounds") and coord.bounds is not None and
+                coord.bounds.dtype == np.float64):
+            if fix:
+                coord.bounds = coord.bounds.astype(np.float32)
+            else:
+                raise TypeError(
+                    "64 bit coord bounds not allowed: {} in {!r}".format(
+                        coord, cube))
+
+
+def check_time_coordinate_metadata(cube):
+    """
+    Function to check time coordinates against the expected standard. The
+    standard for time coordinates is due to technical requirements and if
+    violated the data integrity cannot be guaranteed; so if time coordinates
+    are non-conformant an error is raised.
+
+    Args:
+        cubes (iris.cube.Cube):
+            Cube to be checked
+
+    Raises:
+        ValueError: if any the input cube's time coordinates do not conform
+            to the standard datatypes and units
+    """
+    error_string = ''
+    for time_coord in TIME_METADATA.keys():
+        try:
+            coord = cube.coord(time_coord)
+        except CoordinateNotFoundError:
+            continue
+        reqd_unit = TIME_METADATA[time_coord]["unit"]
+        reqd_dtype = TIME_METADATA[time_coord]["dtype"]
+
+        if not _check_units_and_dtype(coord, reqd_unit, reqd_dtype):
+            msg = ('Coordinate {} does not match required '
+                   'standard (units {}, datatype {})\n')
+            error_string += msg.format(
+                coord.name(), reqd_unit, reqd_dtype)
+
+    # if non-compliance was encountered, raise all messages here
+    if error_string:
+        raise ValueError(error_string)
+
+
+def _construct_object_list(cube, coords):
+    """
+    Construct a list of objects to check
```

Comment:
```suggestion
    Construct a list of objects to check that the datatypes conform to agreed standards.
```
Or something similar?

Reply:
I wouldn't include information in a function docstring about where it's called from.  The function literally constructs a list of objects.  I probably shouldn't even have `to check` in this docstring.

Reply:
Ok that's fine